### PR TITLE
feat: update modified records & complete commits

### PR DESCRIPTION
**Pull request by carlbrugger**
This PR introduces several feature:
- only updates modified records
- set the record message source to `custom-logic`
- fixes an `undefined` bug when fired on record deletion
- fixes a type ("translater" -> "translator")

Closes https://github.com/FlatFilers/flatfile-plugins/issues/270 & https://github.com/FlatFilers/support-triage/issues/722

### CODE REVIEW

File: plugins/record-hook/src/RecordHook.ts [hunk 1]
```diff
@@ -32,33 +37,59 @@ export const RecordHook = async (
...
+        return
+      }
 
-      const recordsUpdates = new RecordTranslater<FlatfileRecord>(
-        batch.records
+      const recordsUpdates = new RecordTranslator<FlatfileRecord>(
+        modifiedRecords
       ).toXRecords()
 
       try {
```
Comment (carlbrugger) [inline plugins/record-hook/src/RecordHook.ts:95]: (next line) `event.update()` will complete the commit, so no need to manually complete it here.

File: plugins/record-hook/src/RecordHook.ts [hunk 2]
```diff
@@ -32,33 +37,59 @@ export const RecordHook = async (
...
+      return
+    }
 
-    // run client defined data hooks
+    // Execute client-defined data hooks
     await asyncBatch(batch.records, handler, options, event)
 
     event.afterAll(async () => {
-      const batch = event.cache.get<FlatfileRecords<any>>('records')
+      const cachedBatch = event.cache.get<FlatfileRecords<any>>('records')
```
[the remark this answers is not in the document]
Comment (carlbrugger) [reply plugins/record-hook/src/RecordHook.ts:81]: This is the latest update. Shouldn't have rename that const. `asyncBatch` updated the batch, (and any other recordHook).

File: plugins/record-hook/src/RecordHook.ts [hunk 3]
```diff
@@ -32,33 +37,59 @@ export const RecordHook = async (
...
     await asyncBatch(batch.records, handler, options, event)
 
     event.afterAll(async () => {
       const batch = event.cache.get<FlatfileRecords<any>>('records')
+      const modifiedRecords = batch.records.filter(hasRecordChanges)
+      if (!modifiedRecords || modifiedRecords.length === 0) {
+        if (options.debug) {
+          console.log('No records modified')
+        }
+        await api.commits.complete(versionId)
```
Comment (ncronquist) [inline plugins/record-hook/src/RecordHook.ts:87]: 💯

File: plugins/record-hook/src/RecordHook.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (carlbrugger) [reply plugins/record-hook/src/RecordHook.ts]: Alternate flow:
- User sets name to "Tom
- recordHook 1 sets name to "john"
- recordHook 2 set name to "Tom"
- recordHook doesn't update record

[the remark this answers is not in the document]
Comment (bangarang) [reply plugins/record-hook/src/RecordHook.ts]: Right, I get that but what if they have multiple RecordHooks that touch the same record in a Listener's Event instance?

[the remark this answers is not in the document]
Comment (bangarang) [reply plugins/record-hook/src/RecordHook.ts]: Sorry, forgot that record.originalValue was set from the server and saved in the cache too 🤦

**Review by bangarang — APPROVED** [review]

**Pull request merged**